clang/include/clang/Driver/Options.td

```diff
@@ -2958,8 +2958,13 @@ defm clangir : BoolFOption<"clangir",
   BothFlags<[], [ClangOption, CC1Option], "">>;
 def emit_cir : Flag<["-"], "emit-cir">, Visibility<[ClangOption, CC1Option]>,
   Group<Action_Group>, HelpText<"Build ASTs and then lower to ClangIR">;
-def emit_core_mlir : Flag<["-"], "emit-core-mlir">, Visibility<[CC1Option]>, Group<Action_Group>,
-  HelpText<"Build ASTs and then lower through ClangIR to core MLIR dialects, emit the .milr file">;
+def emit_mlir_EQ : Joined<["-"], "emit-mlir=">, Visibility<[CC1Option]>, Group<Action_Group>,
+  HelpText<"Build ASTs and then generate/lower to the selected MLIR dialect, emit the .mlir or .cir file. "
+  "Allowed values are `core` for MLIR core dialects and `cir` for ClangIR">,
+  Values<"core,cir">,
```

Comment:
In CIR terms, considering we want to cover DirectToLLVM, ThroughMLIR (core) and CIR, shouldn't this be 3 state: `core,llvm,cir`?

Reply:
I think eventually we will also want to put cir-flat in here, and maybe other variations, but right now these two are the only ones implemented upstream. Technically, you can also lower to the LLVM dialect, but since that's only just now being implemented in the incubator as an output, it isn't done here yet.

Reply:
Makes sense! Thanks